uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -308,28 +310,127 @@ public function getNumberOfWorkingDays()
    *
    * @return int
    */
-  public function getNumberOfWorkingDaysToWork($startDate, $endDate)
-  {
-    if(!CRM_HRLeaveAndAbsences_Validator_Date::isValid($startDate, 'Y-m-d')) {
+  public function getNumberOfWorkingDaysToWork($startDate, $endDate) {
+    if (!CRM_HRLeaveAndAbsences_Validator_Date::isValid($startDate, 'Y-m-d')) {
       throw new InvalidArgumentException('getNumberOfWorkingDaysToWork expects a valid startDate in Y-m-d format');
     }
 
-    if(!CRM_HRLeaveAndAbsences_Validator_Date::isValid($endDate, 'Y-m-d')) {
+    if (!CRM_HRLeaveAndAbsences_Validator_Date::isValid($endDate, 'Y-m-d')) {
       throw new InvalidArgumentException('getNumberOfWorkingDaysToWork expects a valid endDate in Y-m-d format');
     }
 
-    if(strtotime($startDate) < strtotime($this->start_date)) {
+    if (strtotime($startDate) < strtotime($this->start_date)) {
       $startDate = $this->start_date;
     }
 
-    if(strtotime($endDate) > strtotime($this->end_date)) {
+    if (strtotime($endDate) > strtotime($this->end_date)) {
       $endDate = $this->end_date;
     }
 
-    $periodToWork = new CRM_HRLeaveAndAbsences_BAO_AbsencePeriod();
+    $periodToWork             = new CRM_HRLeaveAndAbsences_BAO_AbsencePeriod();
     $periodToWork->start_date = $startDate;
-    $periodToWork->end_date = $endDate;
+    $periodToWork->end_date   = $endDate;
 
     return $periodToWork->getNumberOfWorkingDays();
   }
+
+  /**
+   * Returns the Absence Period previous to this one. That is, the Absence
```

Comment:
Yeah, I agree that doesn't sound like a good idea, but that's how it was designed on the wireframes and you can also see a comment about how it should be used o carry forward calculation on this ticket https://compucorp.atlassian.net/browse/PCHR-918